app/src/main/java/com/amaze/filemanager/utils/ServiceWatcherUtil.java

```diff
@@ -192,4 +213,24 @@ public void run() {
 
         handler.postDelayed(runnable, 0);
     }
+
+    public interface ServiceWatcherInteractionInterface {
```

Comment:
It seems to me that each time this interface is used almost the same code is inside the functions.

Reply:
Depends on the context

Reply:
Abstract class or stating functions then?

Reply:
Abstract class it is, added new commits which reflect the changes.

Reply:
Also, the superclass implements the original interface, why?

Reply:
And the superclass is technically a service, it should not be in utils, it should be under asynchronous.services.

Reply:
>You should use a constructor to set the superclass's fields.

Constructor would be the first thing called while Service would be instantiated. How would I be able to set them at that time.

>Also, the superclass implements the original interface, why?

 To avoid writing same code in all services

>And the superclass is technically a service, it should not be in utils, it should be under asynchronous.services.

Done

Reply:
> To avoid writing same code in all services

How does that work?

Reply:
> Constructor would be the first thing called while Service would be instantiated. How would I be able to set them at that time.

Ok, my mistake, you shouldn't set the variables in the constructor, but somewhere... It is too easy to leave variables initialized, maybe override `onStartCommand()`?